src/snowcli/cli/streamlit/manager.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+import logging
+import typer
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+from snowflake.connector.cursor import SnowflakeCursor
+
+from snowcli.cli.common.sql_execution import SqlExecutionMixin
+from snowcli.cli.snowpark_shared import snowpark_package
+from snowcli.cli.stage.manager import StageManager
+from snowcli.utils import (
+    generate_streamlit_environment_file,
+    generate_streamlit_package_wrapper,
+)
+
+log = logging.getLogger(__name__)
+
+
+class StreamlitManager(SqlExecutionMixin):
+    @staticmethod
+    def get_standard_stage_name(name: str) -> str:
+        # Handle embedded stages
+        if name.startswith("snow://"):
+            return name
+
+        return f"@{name}"
+
+    def list(self) -> SnowflakeCursor:
+        return self._execute_query("show streamlits")
+
+    def description(
+        self, streamlit_name: str
+    ) -> Tuple[SnowflakeCursor, SnowflakeCursor]:
+        description = self._execute_query(f"describe streamlit {streamlit_name}")
+        url = self._execute_query(
+            f"call SYSTEM$GENERATE_STREAMLIT_URL_FROM_NAME('{streamlit_name}')"
+        )
+        return description, url
+
+    def create(
+        self,
+        streamlit_name: str,
+        file: Path,
+        from_stage: str,
+        use_packaging_workaround: bool,
+    ) -> SnowflakeCursor:
+        connection_ctx = self._conn.ctx
+        if from_stage:
+            if "." in from_stage:
+                full_stage_name = from_stage
+            else:
+                full_stage_name = (
+                    f"{connection_ctx.database}.{connection_ctx.schema}.{from_stage}"
+                )
+            standard_page_name = self.get_standard_stage_name(full_stage_name)
+            from_stage_command = f"FROM {standard_page_name}"
+        else:
+            from_stage_command = ""
+        main_file = (
+            "streamlit_app_launcher.py" if use_packaging_workaround else file.name
+        )
+
+        return self._execute_query(
+            f"""
+            create streamlit {streamlit_name}
+            {from_stage_command}
+            MAIN_FILE = '{main_file}'
+            QUERY_WAREHOUSE = {connection_ctx.warehouse};
+
+            alter streamlit {streamlit_name} checkout;
+        """
+        )
+
+    def share(self, streamlit_name: str, to_role: str) -> SnowflakeCursor:
+        return self._execute_query(
+            f"grant usage on streamlit {streamlit_name} to role {to_role}"
+        )
+
+    def drop(self, streamlit_name: str) -> SnowflakeCursor:
+        return self._execute_query(f"drop streamlit {streamlit_name}")
+
+    def deploy(
+        self,
+        streamlit_name: str,
+        file: Path,
+        open_: bool,
```

Comment:
Let's rename it to something more meanigful

Reply:
Renamed to `open_in_browser`